### PR TITLE
Add OTEL propagator

### PR DESCRIPTION
**Pull request by adinauer**
#skip-changelog

## :scroll: Description

Add `SentryPropagator`, register it and use it by default in the Sentry Java Agent JAR.

## :bulb: Motivation and Context

Continue traces (via `sentry-trace` and `baggage` headers) by extracting headers from incoming requests and using that info for Span/Transaction creating as well as injecting headers into outgoing requests.

## :green_heart: How did you test it?
Manually

## :pencil: Checklist

- [ ] I reviewed the submitted code
- [ ] I added tests to verify the changes
- [ ] I updated the docs if needed
- [ ] No breaking changes

### CODE REVIEW

File: sentry-spring/src/main/java/io/sentry/spring/tracing/SentryTracingFilter.java [hunk 1]
```diff
@@ -131,7 +131,8 @@ private ITransaction startTransaction(
                 TransactionNameSource.URL,
                 "http.server",
                 new SentryTraceHeader(sentryTraceHeader),
-                baggage);
+                baggage,
+                null);
```
Comment (romtsn) [inline sentry-spring/src/main/java/io/sentry/spring/tracing/SentryTracingFilter.java:135]: shouldn't this be covered by the method overload without spanId?

**Review by romtsn — APPROVED** [review]

File: sentry-spring/src/main/java/io/sentry/spring/tracing/SentryTracingFilter.java (hunk 1 above)
Comment (adinauer) [reply sentry-spring/src/main/java/io/sentry/spring/tracing/SentryTracingFilter.java:135]: I added another param (`spanId`) to it. The other one doesn't have `baggage` so can't be used here. I didn't want to introduce yet another overload for something that's only used internally anyways. Eventually I'd like to improve this by e.g. introducing a param object that has defaults which can be overriden.

**Review by AbhiPrasad — APPROVED** [review]
I think we might need some more tests, but I'm happy with where this is for now for an alpha!

Let's :shipit:

**Comment by adinauer** [comment]
> I think we might need some more tests

Yes, indeed. It hurts a little to have not tests yet 😢 . Was already on my list of next things to do. Added it as explicit item to #2327 now.

**Pull request merged**